extensions/iam/decentralized-identity/identity-did-crypto/src/main/java/org/eclipse/dataspaceconnector/iam/did/crypto/credentials/VerifiableCredentialFactory.java

```diff
@@ -121,61 +86,48 @@ public static SignedJWT create(PrivateKeyWrapper privateKey, Map<String, String>
     /**
      * Verifies a VerifiableCredential using the issuer's public key
      *
-     * @param verifiableCredential a {@link SignedJWT} that was sent by the claiming party.
-     * @param publicKey The claiming party's public key
-     * @return true if verified, false otherwise
-     */
-    public static boolean verify(SignedJWT verifiableCredential, ECKey publicKey) {
-        try {
-            return verifiableCredential.verify(new ECDSAVerifier(publicKey));
-        } catch (JOSEException e) {
-            throw new CryptoException(e);
-        }
-    }
-
-    /**
-     * Verifies a VerifiableCredential using the issuer's public key
-     *
-     * @param verifiableCredential a {@link SignedJWT} that was sent by the claiming party.
+     * @param jwt       a {@link SignedJWT} that was sent by the claiming party.
      * @param publicKey The claiming party's public key, passed as a {@link PublicKeyWrapper}
+     * @param audience  The intended audience
      * @return true if verified, false otherwise
      */
-    public static boolean verify(SignedJWT verifiableCredential, PublicKeyWrapper publicKey) {
+    public static Result<Void> verify(SignedJWT jwt, PublicKeyWrapper publicKey, String audience) {
+        // verify JWT signature
         try {
-            return verifiableCredential.verify(publicKey.verifier());
+            var verified = jwt.verify(publicKey.verifier());
+            if (!verified) {
+                return Result.failure("Invalid signature");
+            }
         } catch (JOSEException e) {
-            throw new CryptoException(e);
+            // Unable to verify, e.g. the JWS algorithm is not supported
```

Comment:
This comment could be reported as part of the `failure` message

Reply:
@algattik my objective is to avoid the comments, as it should be the last resort, and logging the comment will do the trick, and will make it pretty much impossible to become obsolete

Reply:
Sure, implemented :)